### ISSUE TITLE
Finer grained dependency building

### ISSUE THREAD

**Issue by hsorby** [OpenCMISS/manage#39]
I would like to build just the dependencies for Iron or just the dependencies for Zinc.  Currently I can build all the dependencies (for both Iron and Zinc).

**Comment by chrispbradley** [comment]
Agreed, shouldn't this be flagged from the OC_USE_IRON and OC_USE_ZINC variables? i.e., the external dependencies should be flagged as a dependency from iron/zinc and be added accordingly so that if you turn on or the other off you don't get those dependencies?

**Comment by hsorby** [comment]
This is already done, the dependency building here is if you were just doing a dependency only build so others could link to it from their own builds and not have everyone build their own dependencies.  For HPC's and the like.

**Comment by chrispbradley** [comment]
Then I'm confused as to what you are asking? Is it a make dependencies target? (which should also just look at the use variables to work out what to build?)

**Comment by hsorby** [comment]
I'm asking for a make zinc dependencies target and a make iron dependencies target, the make dependencies target already exists through the configuration.

**Comment by chrispbradley** [comment]
So why doesn't the make dependencies (with OC_USE_IRON off) do what you want?

**Comment by hsorby** [comment]
That leaves zinc to be built.

**Comment by chrispbradley** [comment]
???? I thought this was about building dependencies? I that case why doesn't make opencmiss (with OC_USE_IRON off) do what you want?

**Comment by rondiplomatico** [comment]
guys....

**Comment by rondiplomatico** [comment]
to clarify!

this is currently implemented:
- OC_USE_[IRON|ZINC]
  controls if those components are build. if not, any dependency exclusively relevant to this component will NOT be build.
- OC_DEPENDENCIES_ONLY
  controls if everything BUT iron and zinc is build. if YES, any setting to the OC_USE_[IRON|ZINC] is ignored.

what needs to be done:
make those act the way you'd expect if OC_DEPENDENCIES_ONLY=YES and OC_USE_IRON=NO was set - in that case only the ZINC dependencies would be built.

i've previously added the OC_DEPENDENCIES_ONLY flag for CI purposes related to jenkins builds only, but i suppose it could as well be used for the above scenarios.

@hsorby having an actual _target_ to build those dependencies only is not suitable, as its existence would also coupled to the respective "enablement" of those components. the localconfig is there to select what stuff is build.
